pkg/sentry/cache.go

```diff
@@ -0,0 +1,46 @@
+package sentry
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	defaultCacheTTL = 5 * time.Minute
+)
+
+type Cache struct {
+	mu         sync.Mutex
+	data       map[string]interface{}
+	expiration map[string]time.Time
+}
```

Comment:
Suggestions for maybe future improvements:
Instead of using a map with a mutex, consider using https://pkg.go.dev/sync#Map `sync.Map` is optimised for use-cases like a cache by reducing the number of obtaining locks.